client/src/tests/Carousel.test.tsx

```diff
@@ -0,0 +1,111 @@
+import { render, screen, cleanup, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+import Carousel from '../components/Carousel';
+import componentSmokeTest from './componentSmokeTest';
+
+afterEach(cleanup);
+
+componentSmokeTest({
+  Component: Carousel,
+  props: {
+    imagePaths: [
+      '/general-overview-ex.png',
+      '/key-concepts-ex.png',
+      '/conclusion-ex.png',
+    ],
+    titles: [
+      'General Overview:',
+      'Key Concepts:',
+      'Conclusion:',
+    ],
+    descriptions: [
+      'A top-level breakdown that captures a succinct summary of the uploaded file contents.',
+      'Highlight the top concepts present with a brief summary.',
+      'Finish off your notes with a comprehensive, succinct conclusion.',
+    ],
+  },
+});
+
+const carousel =
+  <Carousel
+    imagePaths={[
+      '/general-overview-ex.png',
+      '/key-concepts-ex.png',
+      '/conclusion-ex.png',
+    ]}
+    titles={[
+      'General Overview:',
+      'Key Concepts:',
+      'Conclusion:',
+    ]}
+    descriptions={[
+      'A top-level breakdown that captures a succinct summary of the uploaded file contents.',
+      'Highlight the top concepts present with a brief summary.',
+      'Finish off your notes with a comprehensive, succinct conclusion.',
+    ]}
+  />;
+
+function testImage(path: string, button?: number) {
```

Comment:
If the second parameter is the number of buttons maybe name it `numButtons`, otherwise it's confusing why `button` is a number.